moor_generator/lib/src/analyzer/sql_queries/query_handler.dart

```diff
@@ -121,6 +127,12 @@ class QueryHandler {
       candidatesForSingleTable.clear();
     }
 
+    if (_foundViews.isNotEmpty) {
+      // For now we're not using the single table optimization when selecting
+      // from views.
+      candidatesForSingleTable.clear();
```

Comment:
Good catch! We could expand the comment to point out that we're not doing this because views don't have data classes yet.

Reply:
Added!